Mentor cell accessibility

SpacyRicochet opened the pull request:
Fixes #35 

**Default text size**

Preferably two cells per row.

**Accessibility maximum text size**

This should increase the label width somehow, probably going to once cell per row.

SpacyRicochet committed:
Add accessibility to mentor cell.

* Set name and country label font to a text style.
* Enable 'Change dynamically'.
* Put views in stackview instead.
* Use cell's content view instead custom view.
* Invalidate layout on trait change.

SpacyRicochet committed:
Adjust estimated item size on larger test styles.

SpacyRicochet committed:
Use font from original design.

ppeelen commented on SwiftIsland/Mentor/MentorCollectionViewCell.swift:
```diff
@@ -13,13 +13,26 @@ class MentorCollectionViewCell: UICollectionViewCell {
   @IBOutlet weak var nameLabel: UILabel!
   @IBOutlet weak var countryLabel: UILabel!
-
+  
```
whitespace, sorry.

SpacyRicochet replied:
I think this resolved itself.

ppeelen commented on SwiftIsland/Mentor/MentorCollectionView.swift:
```diff
@@ -32,7 +32,23 @@ class MentorCollectionView: CardViewController {
+    collectionView?.collectionViewLayout.invalidateLayout()
+    if let layout = collectionView?.collectionViewLayout as? UICollectionViewFlowLayout {
+      if traitCollection.preferredContentSizeCategory.isAccessibilityCategory {
+        layout.estimatedItemSize = CGSize(width: 320, height: 400) // TODO: Not sure how to get a 'one icon per row' layout.
```
What if we separate the calculation on lines 28-32 and reuse it using the `isAccessibilityCategory` flag? It can return a `CGSize` and we might not have to worry about setting the size here?

ppeelen commented on SwiftIsland/Mentor/MentorCollectionViewCell.swift:
```diff
@@ -32,4 +45,11 @@ class MentorCollectionViewCell: UICollectionViewCell {
     }
   }
+  
+  private func setupFonts() {
```
Should this method also be called on `prepareForReuse` in case a reused cell is pulled in without updated fonts?

SpacyRicochet replied:
Unsure actually. Put it there as well.

ppeelen commented on SwiftIsland/Base.lproj/Main.storyboard:
```diff
@@ -1,12 +1,12 @@
+        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="14766.15"/>
         <capability name="Named colors" minToolsVersion="9.0"/>
         <capability name="Safe area layout guides" minToolsVersion="9.0"/>
+        <capability name="collection view cell content view" minToolsVersion="11.0"/>
```
It requires Xcode 11 now, can't release with that yet :(

ppeelen reviewed (changes requested):
Awesome work. Had some thoughts, and it needs to be Xcode 10 compatible unfortunately.

SpacyRicochet committed:
Fix compatibility with Xcode 10. Oops.

Too bad this breaks proper sizing of the collection view cell.

SpacyRicochet committed:
Put font setup in prepare for reuse. Just in case.

SpacyRicochet committed:
Move mentor screens to their own view.

ppeelen commented on SwiftIsland/Helpers/CardViewController.swift:
```diff
@@ -39,7 +39,7 @@ class CardViewController: UIViewController {
     self.visualEffectView = visualEffectView
 
-    guard let vc = storyboard?.instantiateViewController(withIdentifier: "MentorCardViewController") as? MentorCardViewController else { return }
+    guard let vc = UIStoryboard(name: "Mentor", bundle: nil).instantiateViewController(withIdentifier: "MentorCardViewController") as? MentorCardViewController else { return }
```
Nice idea to separate it! =)

SpacyRicochet committed:
Looks better, but definitely not according to design.

SpacyRicochet commented:
I'm at the 'Oh well 🤷‍♂' stage now. This is why Dynamic Type is so hard to get right in apps.

![cells-not-resizing-correctly](https://user-images.githubusercontent.com/904596/60114387-08bb3d80-9774-11e9-9288-4a9e7e03942f.gif)

SpacyRicochet committed:
Boooh, self-sizing! Booh! 👻

This puts the name and country in the same label, which simplifies the internal calculations. It also drops self-sizing in favor of artisanally crafted calculations! 👌

SpacyRicochet commented:
Final commit brings us to the following result;

**Default text style**

![Simulator Screen Shot - iPhone 5s - 2019-06-25 at 21 15 02](https://user-images.githubusercontent.com/904596/60126636-bafffe80-978e-11e9-9819-3b19eb2ac184.png)

**AXXXL text style**

![Simulator Screen Shot - iPhone 5s - 2019-06-25 at 21 15 18](https://user-images.githubusercontent.com/904596/60126658-c2bfa300-978e-11e9-9afe-3c3720cb8b16.png)

![Simulator Screen Shot - iPhone 5s - 2019-06-25 at 21 18 38](https://user-images.githubusercontent.com/904596/60126685-cfdc9200-978e-11e9-8f29-6521f1e17ab6.png)
